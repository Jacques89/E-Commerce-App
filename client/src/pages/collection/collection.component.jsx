import React from 'react'
import { connect } from 'react-redux';

import CollectionItemContainer from '../../components/collection-item/collection-item.container';
import { selectCollection } from '../../redux/shop/shop.selectors';

import { CollectionPageContainer, CollectionTitle, CollectionItemsDiv } from './collection.styles';

const CollectionPage = ({ collection: { title, items} }) => (
    <CollectionPageContainer>
        <CollectionTitle>{title}</CollectionTitle>
        <CollectionItemsDiv>
            {
                items.map(item => <CollectionItemContainer key={item.id} item={item} />)
            }
        </CollectionItemsDiv>
    </CollectionPageContainer>
);

const mapStateToProps = (state, ownProps) => ({
    collection: selectCollection(ownProps.match.params.collectionId)(state)
})

export default connect(mapStateToProps)(CollectionPage);