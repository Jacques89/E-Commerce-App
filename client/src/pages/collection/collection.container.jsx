import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'
import { compose } from 'redux'

import { selectIsCollectionsLoaded } from 'redux/shop/shop.selectors'
import WithSpinner from 'components/Loading/with-spinner/with-spinner.component'
import CollectionPage from './collection.component'

const mapStateToProps = createStructuredSelector({
    isLoading: selectIsCollectionsLoaded
})

const CollectionPageContainer = compose(connect(mapStateToProps), WithSpinner)(CollectionPage)

export default CollectionPageContainer
