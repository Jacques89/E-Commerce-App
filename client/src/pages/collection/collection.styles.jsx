import styled from 'styled-components'

export const CollectionPageContainer = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
`

export const CollectionTitle = styled.h2`
    font-size: 2.375rem;
    margin: 0 auto 1.875rem;
`

export const CollectionItemsDiv = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-gap: 0.938rem;
    & > div {
        margin-bottom: 1.875rem;
    }

    @media screen and (max-width: 800px) {
        grid-template-columns: 1fr;
        grid-gap: 0.938rem;
    }

    @media screen and (max-width: 554px) {
        grid-template-columns: 1fr;
        grid-gap: 0.938rem;
        width: 40%;
    }
`

CollectionItemsDiv.displayName = 'CollectionItemsStyle'
