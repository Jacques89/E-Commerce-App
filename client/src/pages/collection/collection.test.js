import React from 'react'
import { mount } from 'enzyme'
import configureMockStore from 'redux-mock-store'

import { CollectionPage } from './collection.component'
import CollectionItem from '../../components/Collection/collection-item/collection-item.component'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'

describe('CollectionPage', () => {
    let wrapper
    let mockItems = [{ id: 1 }, { id: 2 }, { id: 3 }]

    const mockStore = configureMockStore()
    const store = mockStore({})

    beforeEach(() => {
        const mockCollection = {
            items: mockItems,
            title: 'Test'
        }

        wrapper = mount(
            <Provider store={store}>
                <Router>
                    <CollectionPage collection={mockCollection} />
                </Router>
            </Provider>
        )
    })

    it('should render the CollectionPage component', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should render the same number of CollectionItems as collection array', () => {
        expect(wrapper.find(CollectionItem).length).toBe(mockItems.length)
    })
})
