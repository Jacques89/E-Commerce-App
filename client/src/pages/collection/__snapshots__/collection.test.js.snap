// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage should render the CollectionPage component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <CollectionPage
        collection={
          Object {
            "items": Array [
              Object {
                "id": 1,
              },
              Object {
                "id": 2,
              },
              Object {
                "id": 3,
              },
            ],
            "title": "Test",
          }
        }
      >
        <styled.div>
          <div
            className="sc-iBPTik jLHUJN"
          >
            <styled.h2>
              <h2
                className="sc-fubCzh kggWwV"
              >
                Test
              </h2>
            </styled.h2>
            <CollectionItemsStyle>
              <div
                className="sc-pGacB dCTXWU"
              >
                <Connect(withRouter(CollectionItem))
                  item={
                    Object {
                      "id": 1,
                    }
                  }
                  key="1"
                >
                  <withRouter(CollectionItem)
                    addItem={[Function]}
                    item={
                      Object {
                        "id": 1,
                      }
                    }
                  >
                    <CollectionItem
                      addItem={[Function]}
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      item={
                        Object {
                          "id": 1,
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                    >
                      <CollectionItemStyles>
                        <div
                          className="sc-gsTEea gpvzcy"
                        >
                          <BackgroundImageStyles
                            className="image"
                            onClick={[Function]}
                          >
                            <div
                              className="sc-hKgJUU katQCk image"
                              onClick={[Function]}
                            />
                          </BackgroundImageStyles>
                          <CollectionFooterStyles>
                            <div
                              className="sc-eCstlR hXZTWx"
                            >
                              <NameSpanStyles>
                                <span
                                  className="sc-jSgvzq cpdSLY"
                                />
                              </NameSpanStyles>
                              <PriceSpanStyles>
                                <span
                                  className="sc-gKseQn edKNxZ"
                                >
                                  €
                                </span>
                              </PriceSpanStyles>
                            </div>
                          </CollectionFooterStyles>
                          <AddButtonStyles
                            inverted={true}
                            onClick={[Function]}
                          >
                            <CustomButton
                              className="sc-dlfnuX eznRbH"
                              inverted={true}
                              onClick={[Function]}
                            >
                              <styled.button
                                className="sc-dlfnuX eznRbH"
                                inverted={true}
                                onClick={[Function]}
                              >
                                <button
                                  className="sc-bdfBQB doWsIP sc-dlfnuX eznRbH"
                                  onClick={[Function]}
                                >
                                  Add to cart
                                </button>
                              </styled.button>
                            </CustomButton>
                          </AddButtonStyles>
                        </div>
                      </CollectionItemStyles>
                    </CollectionItem>
                  </withRouter(CollectionItem)>
                </Connect(withRouter(CollectionItem))>
                <Connect(withRouter(CollectionItem))
                  item={
                    Object {
                      "id": 2,
                    }
                  }
                  key="2"
                >
                  <withRouter(CollectionItem)
                    addItem={[Function]}
                    item={
                      Object {
                        "id": 2,
                      }
                    }
                  >
                    <CollectionItem
                      addItem={[Function]}
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      item={
                        Object {
                          "id": 2,
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                    >
                      <CollectionItemStyles>
                        <div
                          className="sc-gsTEea gpvzcy"
                        >
                          <BackgroundImageStyles
                            className="image"
                            onClick={[Function]}
                          >
                            <div
                              className="sc-hKgJUU katQCk image"
                              onClick={[Function]}
                            />
                          </BackgroundImageStyles>
                          <CollectionFooterStyles>
                            <div
                              className="sc-eCstlR hXZTWx"
                            >
                              <NameSpanStyles>
                                <span
                                  className="sc-jSgvzq cpdSLY"
                                />
                              </NameSpanStyles>
                              <PriceSpanStyles>
                                <span
                                  className="sc-gKseQn edKNxZ"
                                >
                                  €
                                </span>
                              </PriceSpanStyles>
                            </div>
                          </CollectionFooterStyles>
                          <AddButtonStyles
                            inverted={true}
                            onClick={[Function]}
                          >
                            <CustomButton
                              className="sc-dlfnuX eznRbH"
                              inverted={true}
                              onClick={[Function]}
                            >
                              <styled.button
                                className="sc-dlfnuX eznRbH"
                                inverted={true}
                                onClick={[Function]}
                              >
                                <button
                                  className="sc-bdfBQB doWsIP sc-dlfnuX eznRbH"
                                  onClick={[Function]}
                                >
                                  Add to cart
                                </button>
                              </styled.button>
                            </CustomButton>
                          </AddButtonStyles>
                        </div>
                      </CollectionItemStyles>
                    </CollectionItem>
                  </withRouter(CollectionItem)>
                </Connect(withRouter(CollectionItem))>
                <Connect(withRouter(CollectionItem))
                  item={
                    Object {
                      "id": 3,
                    }
                  }
                  key="3"
                >
                  <withRouter(CollectionItem)
                    addItem={[Function]}
                    item={
                      Object {
                        "id": 3,
                      }
                    }
                  >
                    <CollectionItem
                      addItem={[Function]}
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      item={
                        Object {
                          "id": 3,
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                    >
                      <CollectionItemStyles>
                        <div
                          className="sc-gsTEea gpvzcy"
                        >
                          <BackgroundImageStyles
                            className="image"
                            onClick={[Function]}
                          >
                            <div
                              className="sc-hKgJUU katQCk image"
                              onClick={[Function]}
                            />
                          </BackgroundImageStyles>
                          <CollectionFooterStyles>
                            <div
                              className="sc-eCstlR hXZTWx"
                            >
                              <NameSpanStyles>
                                <span
                                  className="sc-jSgvzq cpdSLY"
                                />
                              </NameSpanStyles>
                              <PriceSpanStyles>
                                <span
                                  className="sc-gKseQn edKNxZ"
                                >
                                  €
                                </span>
                              </PriceSpanStyles>
                            </div>
                          </CollectionFooterStyles>
                          <AddButtonStyles
                            inverted={true}
                            onClick={[Function]}
                          >
                            <CustomButton
                              className="sc-dlfnuX eznRbH"
                              inverted={true}
                              onClick={[Function]}
                            >
                              <styled.button
                                className="sc-dlfnuX eznRbH"
                                inverted={true}
                                onClick={[Function]}
                              >
                                <button
                                  className="sc-bdfBQB doWsIP sc-dlfnuX eznRbH"
                                  onClick={[Function]}
                                >
                                  Add to cart
                                </button>
                              </styled.button>
                            </CustomButton>
                          </AddButtonStyles>
                        </div>
                      </CollectionItemStyles>
                    </CollectionItem>
                  </withRouter(CollectionItem)>
                </Connect(withRouter(CollectionItem))>
              </div>
            </CollectionItemsStyle>
          </div>
        </styled.div>
      </CollectionPage>
    </Router>
  </BrowserRouter>
</Provider>
`;
